src/live_restore/live_restore_server.c

```diff
@@ -416,14 +434,15 @@ __wt_live_restore_server_destroy(WT_SESSION_IMPL *session)
     WTI_LIVE_RESTORE_SERVER *server = S2C(session)->live_restore_server;
 
     /*
-     * If we didn't create a live restore file system or the server there is nothing to do, it is
-     * rare, but possible, to arrive here with the flag set and a NULL server. This situation
-     * happens when an error is encountered after the file system initialization but before the
-     * server is created.
+     * If we didn't create the background migration server there is nothing to do. It is rare, but
+     * possible, to arrive here with the flag set and a NULL server. This situation arises when an
+     * error is encountered during the server set up.
      */
-    if (!F_ISSET(S2C(session), WT_CONN_LIVE_RESTORE_FS) || server == NULL)
+    if (!F_ISSET(S2C(session), WT_CONN_LIVE_RESTORE_MIGRATION_RUNNING) || server == NULL)
```

Comment:
Could we just check if `server == NULL`? The existence of a server means we need to destroy it.

Reply:
Fixed in [da437c9](https://github.com/wiredtiger/wiredtiger/pull/11797/commits/da437c98eb0011de7f62533c0dc576e8d46e06a0). We also needed to change the `__wt_free` call to free the pointer on the connection, not free the local variable that contains the same pointer.